src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeftTestSame.java

```diff
@@ -0,0 +1,210 @@
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+/*
+ * Config:
+ * option = same
+ * tokens = { LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, LITERAL_IF,
+ *            LITERAL_ELSE, LITERAL_FOR, LITERAL_WHILE, LITERAL_DO, ANNOTATION_DEF, ENUM_DEF }
+ */
+class InputRightCurlyLeftTestSame
+{
+    /** @see test method **/
+    int foo() throws InterruptedException
+    {
+        int x = 1;
+        int a = 2;
+        while (true)
+        {
+            try
+            {
+                if (x > 0)
+                {
+                    break;
+                } // violation
+                else if (x < 0) {
+                    ;
+                } // violation
+                else
+                {
+                    break;
+                }
+                switch (a)
+                {
+                case 0:
+                    break;
+                default:
+                    break;
+                }
+            } // violation
+            catch (Exception e)
+            {
+                break;
+            } // violation
+            finally
+            {
+                break;
+            }
+        }
+
+        synchronized (this)
+        {
+            do
+            {
+                x = 2;
+            } while (x == 2); // ok
+        }
+
+        this.wait(666
+                 ); // Bizarre, but legal
+
+        for (int k = 0; k < 1; k++)
+        {
+            String innerBlockVariable = "";
+        }
+
+        // test input for bug reported by Joe Comuzzi
+        if (System.currentTimeMillis() > 1000)
+            return 1;
+        else
+            return 2;
+    }
+
+    // Test static initialiser
+    static
+    {
+        int x = 1; // should not require any javadoc
+    }
+
+
+
+    public enum GreetingsEnum
+    {
+        HELLO,
+        GOODBYE
+    }; // ok
```

Comment:
it says ok, but violations are expected according to the test case, please check also lines 91, 186, and 187

Reply:
Done!